array_api_strict/tests/test_validation.py

```diff
@@ -1,11 +1,9 @@
-from typing import Callable
-
 import pytest
 
 import array_api_strict as xp
 
 
-def p(func: Callable, *args, **kwargs):
+def p(func, *args, **kwargs):
```

Comment:
Something like 

```suggestion
def p(func: Callable[..., object], *args: object, **kwargs: object) -> pytest.param:
```

seems just as easy to me 🤷🏻.

But that being said, I'm the type that would even stub his own toe.

Reply:
Hard no from me. This is a test module.

Reply:
> This is a test module.

Yea I noticed that.

> Hard no from me.

Why?

Reply:
type annotations in test modules unnecessarily increase attrition and code verbosity without giving any material benefit to code maintainability or robustness. A pathological example of that is array-api-extra.

Reply:
I don't really agree with that. 

Tests have to be maintained as well, and annotations make it easier for the maintainer to see what the *intent* of the tests is. 

In case of test failures, it can be very helpful to see the annotations printed in the error output, because in many cases it is helpful to know the *intent* of the tests when you are figuring out when it fails.

So I guess I view type annotations in tests as a concise yet precise form of documentation, specifically meant to make our own (i.e. the maintainers) lives a bit easier.

Either way, do whatever floats your boat. I just know that it has helped in my projects, so I think it can help here as well.